### DISCUSSION TITLE
GD32F303 microcontroler to Ender3 V2

### DISCUSSION THREAD

**Discussion by niaggeli** [CrealityOfficial/Creality_Sonic_Pad#25] [General]
Hi creality made some changes using another microcontroler for ender 3 v2 .anymore they are  using GD32F303 in order of  stm32f103, but the  result is that we cant install Klipper to our machines. Even some options from Marlin firmwares some times we cant use them correct. Does anybody know what we should do to use Klipper to our machines?

**Comment by thewestwest** [comment]
I have GD32F303 on a 4.2.2 board. I have klipper connected to moonraker/mainsail - it's taking commands and seems to be working although I am not finished with all of the config yet; also 8 am doing this thru Linux on a desktop not via a raspberry ..for me, I seem to have it working with the following: in make menuconfig select that you to enable extra low-level options then type Q and Y to close and save. reopen make menuconfig again and select stm32f103, select disable swd, and select 28KiB bootloader. change the communication to USB on (PA11/PA12); type Q and Y again to save and exit, then run 'make' in terminal. flash the firmware it generates to your board. connect printer via usb and  run 'ls /dev/serial/by-id/*' to get the address that you will use in [mcu] section. that got my steppers responding and now i'm just filling  in the rest of the config and hoping for the best - so far so good. good luck

**Comment by NeilLB13** [comment]
Please excuse me for any noob mistakes; it's my first post and I never expected to have to get into this at all.
I have a Creality Ender 3 V2 Neo, and have bought a Sonic Pad to use with it. Unfortunately it wasn't simply "plug and play" as expected.
When I follow the flashing instructions on the Sonic Pad, it bricks my printer, and I have to re-flash it to V1.1.7 to get it to work again.
It looks as though the reason for this is that it's trying to flash the memory as if it had a STM chipset, but having been inside the case it's a V4.2.2 board with a GD32F303 (the rest of the writing on it is "RET6", "AUJ5631", "AJ2217", "GigaDevice" and "ARM").
I've tried a support request with Creality but they're being no help at all (they couldn't even diagnose the WiFi problem that I'd been having, even though the fix was just to remove spaces from the SSID).
I guess the easiest fix would be if anyone would be prepared to share a Klipper.bin file that works on the Ender 3 V2 Neo, which would presumably then allow the Sonic Pad to communicate with the printer.
But failing that, could anyone post instrucitons or talk me through the process of getting it to work please?
Thanks & regards,
Neil.
